projects/traversal/src/lib/traverser.ts

```diff
@@ -323,18 +323,24 @@ export class Traverser {
         };
     }
 
-    getQueryParams(): Observable<{ [key: string]: string }> {
+    getQueryParams(): Observable<{ [key: string]: string[]| string }> {
         return this.target.pipe(
```

Comment:
I'd prefer to return always an array of string but I see that it wouldn't be backward compatible.

Reply:
when using `getQueryParams` we will never know in advanced what's the type of a given parameter, the compiler will complain if we do not handle that manually on every call:
```
this.traverser.getQueryParams().subscribe(params => {
     const token: string = params.token; // ERROR because token is a string, not a string | string[]
});
```
Most part of time, parameters will be just strings, so polluting the general usage with manual type checking is painfull.

What about:
```
getQueryParams<T = string>(): Observable<{ [key: string]: T }>
```
?
that way we can call `this.traverser.getQueryParams()` as usual and get strings, or we can do ```this.traverser.getQueryParams<string | string[]>()``` when we know we will get list of strings.

Reply:
That's a good idea, it gives the flexibility I was looking for.

Reply:
Done